src/main/java/net/fabricmc/loom/task/MigrateMappingsTask.java

```diff
@@ -73,27 +71,8 @@ public void doTask() throws Throwable {
 		if (inputDir == null) inputDir = "src/main/java";
 		String outputDir = (String) properties.get("outputDir");
 		if (outputDir == null) outputDir = "remappedSrc";
-		String officialMappingsVersion = (String) properties.get("targetMappings");
-		String localMappingsPath = (String) properties.get("targetLocalMappings");
-
-		if (officialMappingsVersion != null && localMappingsPath != null) {
-			throw new IllegalArgumentException("targetMappings and targetLocalMappings are mutually exclusive;"
-							+ " you either specify an official yarn mappings version with targetMappings, "
-							+ "or a local one with targetLocalMappings.");
-		}
-
-		if (officialMappingsVersion == null && localMappingsPath == null) {
-			throw new IllegalArgumentException("You must specify a new mappings version with -PtargetMappings (or local mappings with -PtargetLocalMappings).");
-		}
-
-		boolean useLocalMappings = localMappingsPath != null;
-
-		if (useLocalMappings && !Files.exists(Paths.get(localMappingsPath))) {
-			throw new IllegalArgumentException("Can't find local mappings in specified location: " + localMappingsPath);
-		}
-
-		String targetMappingsName = useLocalMappings ? localMappingsPath : officialMappingsVersion;
 
+		File mappings = project.getConfigurations().detachedConfiguration(project.getDependencies().module(properties.get("mappings"))).getSingleFile();
```

Comment:
1. `properties.get("mappings"))` needs a null check
2. Would be nice to at least have a way to only need to specify the yarn version, instead of net.fabricmc.yarn.<version>, since that's 99.9999% of the use case.